### PR TITLE
evince: update to 45.0

### PR DESCRIPTION
**Pull request by pokoli**
Hi, 

I took the time to update evince package to 45.0 version. Note that this is my first PR in this repository, so probably I'm missed something in the process. Let me know in such case so I can fix it. 

In the update process I removed all patches that seemed outdated and just keep the ones need to make the code compyle on msys2. For that I did:

- Remove some parts of the code that used the `fnctl` function which is not available on the build. For that I updated the current patch that already had some issue. 
- Remove the `pdf_document_load_fn` function which caused a compiling error. 

This allowed me to build the package.

### CODE REVIEW

File: mingw-w64-evince/PKGBUILD [hunk 1]
```diff
@@ -28,35 +28,28 @@ depends=("${MINGW_PACKAGE_PREFIX}-glib2"
...
-             "${MINGW_PACKAGE_PREFIX}-pkgconf"
+             "${MINGW_PACKAGE_PREFIX}-desktop-file-utils"
              "${MINGW_PACKAGE_PREFIX}-gettext-tools"
              "${MINGW_PACKAGE_PREFIX}-gobject-introspection"
              "${MINGW_PACKAGE_PREFIX}-gtk-doc"
              "${MINGW_PACKAGE_PREFIX}-itstool"
+             "${MINGW_PACKAGE_PREFIX}-libhandy"
+             "${MINGW_PACKAGE_PREFIX}-meson"
+             "${MINGW_PACKAGE_PREFIX}-ninja"
+             "${MINGW_PACKAGE_PREFIX}-pkgconf"
```
Comment (MehdiChinoune) [inline mingw-w64-evince/PKGBUILD:39]: packages were ordered as: compilers, build-systems, build/configure tools (ninja, pkgconf...), general tools (like gettext-tools), buildtime libraries, documentation genarators/tools.
It's just a personal preference, so not a big deal. but as a new contributor, just leave as it is.

**Comment by pokoli** [comment]
I've seen the build is failing when extracting the evince code with the following error: 

`evince-45.0/data/icons/hicolor: Can't create 'evince-45.0/data/icons/hicolor': File name too long`

I also had the same error in my local build with I fixed running: 

`$ export MSYS=winsymlinks:lnk`

Before running the build. I took this soultion from https://github.com/msys2/MSYS2-packages/issues/1216
Not sure if this is something I need to add to the package build or must be fixed in a generic manner.

File: mingw-w64-evince/PKGBUILD (hunk 1 above)
Comment (pokoli) [reply mingw-w64-evince/PKGBUILD]: I thought the order was alphabetical so I followed this order. 
I moved the compilers back to top, so now it follows the same issue as before. 

Thanks for your fast feedback here!

**Comment by MehdiChinoune** [comment]
We fixed the extaction issue by adding the archive to `noextract=()` and then extract it manually in `prepare()` step, take a look at mingw-w64-clang/PKGBUILD as an example

**Comment by pokoli** [comment]
Thanks @MehdiChinoune! I kindly appreciate your fast feedback here its allowed to fix the build in just one hour. You made my day :partying_face:

**Comment by hmartinez82** [comment]
Failing to build with CLANGARM64 unfortunately, and I don't even understand the error :(
```
[1/3] "C:/msys64/clangarm64/bin/meson" "--internal" "exe" "--unpickle" "C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/meson-private/meson_exe_g-ir-scanner.exe_d91d6ee028cd49a0150b619aa1239c1ea005748c.dat"
FAILED: libview/EvinceView-3.0.gir
"C:/msys64/clangarm64/bin/meson" "--internal" "exe" "--unpickle" "C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/meson-private/meson_exe_g-ir-scanner.exe_d91d6ee028cd49a0150b619aa1239c1ea005748c.dat"
while executing ['C:/msys64/clangarm64/bin/g-ir-scanner.exe', '--quiet', '--no-libtool', '--namespace=EvinceView', '--nsversion=3.0', '--warn-all', '--output', 'libview/EvinceView-3.0.gir', '-DEVINCE_COMPILATION', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/libview', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libview', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/libdocument/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libdocument/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/cut-n-paste/synctex/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/cut-n-paste/synctex/.', '--filelist=C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libview/libevview3-3.dll.p/EvinceView_3.0_gir_filelist', '--include=Gdk-3.0', '--include=GdkPixbuf-2.0', '--include=Gio-2.0', '--include=GLib-2.0', '--include=GObject-2.0', '--include=Gtk-3.0', '--include-uninstalled=libdocument/EvinceDocument-3.0.gir', '--symbol-prefix=ev', '--identifier-prefix=Ev', '--pkg-export=evince-view-3.0', '--cflags-begin', '-DHAVE_CONFIG_H', '-D_WIN32_WINNT=0x0500', '-DHAVE_CONFIG_H', '-D_WIN32_WINNT=0x0500', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/libdocument/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libdocument/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/cut-n-paste/synctex/.', '-IC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/cut-n-paste/synctex/.', '-IC:/msys64/clangarm64/include/gspell-1', '-IC:/msys64/clangarm64/include/gtk-3.0', '-IC:/msys64/clangarm64/include/pango-1.0', '-IC:/msys64/clangarm64/include/harfbuzz', '-IC:/msys64/clangarm64/include/freetype2', '-IC:/msys64/clangarm64/include/cairo', '-IC:/msys64/clangarm64/include/gdk-pixbuf-2.0', '-IC:/msys64/clangarm64/include/atk-1.0', '-IC:/msys64/clangarm64/include/enchant-2', '-IC:/msys64/clangarm64/include/glib-2.0', '-IC:/msys64/clangarm64/lib/glib-2.0/include', '-IC:/msys64/clangarm64/include/webp', '-DLIBDEFLATE_DLL', '-IC:/msys64/clangarm64/include/libpng16', '-IC:/msys64/clangarm64/include/pixman-1', '-IC:/msys64/clangarm64/include/fribidi', '-IC:/msys64/clangarm64/include/gstreamer-1.0', '-IC:/msys64/clangarm64/include/orc-0.4', '-IC:/msys64/clangarm64/include/libxml2', '-IC:/msys64/clangarm64/include/gobject-introspection-1.0', '-D__USE_MINGW_ANSI_STDIO=1', '-D__USE_MINGW_ANSI_STDIO=1', '--cflags-end', '--add-include-path=C:/msys64/clangarm64/share/gir-1.0', '--add-include-path=C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libdocument', '-LC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libview', '--library', 'evview3', '-LC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libdocument', '-LC:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/libdocument', '--library', 'evdocument3', '--extra-library=gspell-1', '--extra-library=gtk-3', '--extra-library=gdk-3', '--extra-library=z', '--extra-library=gdi32', '--extra-library=imm32', '--extra-library=shell32', '--extra-library=ole32', '--extra-library=uuid', '--extra-library=winmm', '--extra-library=dwmapi', '--extra-library=setupapi', '--extra-library=cfgmgr32', '--extra-library=hid', '--extra-library=winspool', '--extra-library=comctl32', '--extra-library=comdlg32', '--extra-library=pangowin32-1.0', '--extra-library=harfbuzz', '--extra-library=pangocairo-1.0', '--extra-library=pango-1.0', '--extra-library=atk-1.0', '--extra-library=cairo', '--extra-library=cairo-gobject', '--extra-library=gdk_pixbuf-2.0', '--extra-library=gio-2.0', '--extra-library=glib-2.0', '--extra-library=intl', '--extra-library=gobject-2.0', '--extra-library=enchant-2', '--extra-library=gstbase-1.0', '--extra-library=gstreamer-1.0', '--extra-library=gstvideo-1.0', '--extra-library=gthread-2.0', '--extra-library=gmodule-2.0', '--extra-library=xml2', '--extra-library=m', '--extra-library=shlwapi', '--extra-library=girepository-1.0', '--sources-top-dirs', 'C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/evince-45.0/', '--sources-top-dirs', 'C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/']
--- stdout ---

--- stderr ---
C:\msys64\clangarm64\include\glib-2.0\glib\gutils.h:378: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 0); if (nth_bit < -1) G_PASTE (_g_boolean_var_, 0) = 1; else G_PASTE (_g_boolean_var_, 0) = 0; G_PASTE (_g_boolean_var_, 0); }), 0)))' at '0'
C:\msys64\clangarm64\include\glib-2.0\glib\gutils.h:393: syntax error, unexpected INTEGER, expecting identifier in '  if (nth_bit < 0 || (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 1); if (nth_bit > 4 * 8) G_PASTE (_g_boolean_var_, 1) = 1; else G_PASTE (_g_boolean_var_, 1) = 0; G_PASTE (_g_boolean_var_, 1); }), 0)))' at '1'
C:\msys64\clangarm64\include\glib-2.0\glib\gutils.h:408: syntax error, unexpected INTEGER, expecting identifier in '  return (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 2); if (number) G_PASTE (_g_boolean_var_, 2) = 1; else G_PASTE (_g_boolean_var_, 2) = 0; G_PASTE (_g_boolean_var_, 2); }), 1)) ?' at '2'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:198: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 3); if (gstring != ((void*)0) && gstring->len + 1 < gstring->allocated_len) G_PASTE (_g_boolean_var_, 3) = 1; else G_PASTE (_g_boolean_var_, 3) = 0; G_PASTE (_g_boolean_var_, 3); }), 1)))' at '3'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:220: syntax error, unexpected INTEGER, expecting identifier in '  if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 4); if (gstring == ((void*)0)) G_PASTE (_g_boolean_var_, 4) = 1; else G_PASTE (_g_boolean_var_, 4) = 0; G_PASTE (_g_boolean_var_, 4); }), 0))' at '4'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:223: syntax error, unexpected INTEGER, expecting identifier in '  if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 5); if (val == ((void*)0)) G_PASTE (_g_boolean_var_, 5) = 1; else G_PASTE (_g_boolean_var_, 5) = 0; G_PASTE (_g_boolean_var_, 5); }), 0))' at '5'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:231: syntax error, unexpected INTEGER, expecting identifier in '  if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 6); if (gstring->len + len_unsigned < gstring->allocated_len) G_PASTE (_g_boolean_var_, 6) = 1; else G_PASTE (_g_boolean_var_, 6) = 0; G_PASTE (_g_boolean_var_, 6); }), 1)))' at '6'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:234: syntax error, unexpected IF in '      if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 7); if (val + len_unsigned <= end || val > end + len_unsigned) G_PASTE (_g_boolean_var_, 7) = 1; else G_PASTE (_g_boolean_var_, 7) = 0; G_PASTE (_g_boolean_var_, 7); }), 1)))' at 'if'
C:\msys64\clangarm64\include\glib-2.0\glib\gstring.h:234: syntax error, unexpected INTEGER, expecting identifier in '      if ((__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 7); if (val + len_unsigned <= end || val > end + len_unsigned) G_PASTE (_g_boolean_var_, 7) = 1; else G_PASTE (_g_boolean_var_, 7) = 0; G_PASTE (_g_boolean_var_, 7); }), 1)))' at '7'
C:\msys64\clangarm64\include\glib-2.0\gobject\gobject.h:819: syntax error, unexpected INTEGER, expecting identifier in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 8); if ((((g_type_check_instance_is_fundamentally_a ((GTypeInstance*) ((weak_pointer)), (((GType) ((20) << (2))))))))) G_PASTE (_g_boolean_var_, 8) = 1; else G_PASTE (_g_boolean_var_, 8) = 0; G_PASTE (_g_boolean_var_, 8); }), 1)) ; else g_assertion_message (((gchar*) 0), "C:/msys64/clangarm64/include/glib-2.0/gobject/gobject.h", 819, ((const char*) (__func__)), "'" "G_IS_OBJECT (weak_pointer)" "' should be TRUE"); } while (0);' at '8'
C:\msys64\clangarm64\include\glib-2.0\gobject\gobject.h:819: syntax error, unexpected ')', expecting identifier or '(' in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 8); if ((((g_type_check_instance_is_fundamentally_a ((GTypeInstance*) ((weak_pointer)), (((GType) ((20) << (2))))))))) G_PASTE (_g_boolean_var_, 8) = 1; else G_PASTE (_g_boolean_var_, 8) = 0; G_PASTE (_g_boolean_var_, 8); }), 1)) ; else g_assertion_message (((gchar*) 0), "C:/msys64/clangarm64/include/glib-2.0/gobject/gobject.h", 819, ((const char*) (__func__)), "'" "G_IS_OBJECT (weak_pointer)" "' should be TRUE"); } while (0);' at ')'
C:\msys64\clangarm64\include\glib-2.0\gobject\gobject.h:822: syntax error, unexpected INTEGER, expecting identifier in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 9); if ((weak_pointer) == ((void*)0)) G_PASTE (_g_boolean_var_, 9) = 1; else G_PASTE (_g_boolean_var_, 9) = 0; G_PASTE (_g_boolean_var_, 9); }), 1)) ; else g_assertion_message (((gchar*) 0), "C:/msys64/clangarm64/include/glib-2.0/gobject/gobject.h", 822, ((const char*) (__func__)), "'" "weak_pointer" "' should be NULL"); } while (0);' at '9'
C:\msys64\clangarm64\include\glib-2.0\gobject\gobject.h:822: syntax error, unexpected ')', expecting identifier or '(' in '  do { if (__builtin_expect (__extension__ ({ int G_PASTE (_g_boolean_var_, 9); if ((weak_pointer) == ((void*)0)) G_PASTE (_g_boolean_var_, 9) = 1; else G_PASTE (_g_boolean_var_, 9) = 0; G_PASTE (_g_boolean_var_, 9); }), 1)) ; else g_assertion_message (((gchar*) 0), "C:/msys64/clangarm64/include/glib-2.0/gobject/gobject.h", 822, ((const char*) (__func__)), "'" "weak_pointer" "' should be NULL"); } while (0);' at ')'
clang: warning: argument unused during compilation: '-shared' [-Wunused-command-line-argument]
Command '['C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/tmp-introspectqotnvh0d/EvinceView-3.0.exe', '--introspect-dump=C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/tmp-introspectqotnvh0d/functions.txt,C:/Dev/Github/MINGW-packages/mingw-w64-evince/src/build-CLANGARM64/tmp-introspectqotnvh0d/dump.xml']' returned non-zero exit status 3221225781.

ninja: build stopped: subcommand failed.
```

**Comment by pokoli** [comment]
Do I need to do something to make the PR move forward?

**Comment by Biswa96** [comment]
I just realized that this does not work, neither did the older version. Does this work for your case?

**Comment by Biswa96** [comment]
The error output is as following.

```
Unable to open document “file:///path/to/file.pdf”.
Source _rels/.rels not found in archive
```

**Comment by hmartinez82** [comment]
I'm getting the same in CLANGARM64 
image

**Pull request merged**